src/Beacon.jsx

```diff
@@ -1,24 +1,21 @@
 import React, { PropTypes } from 'react';
-import Portal from 'react-portal';
 import sha1 from 'sha1';
 import withClickOutside from 'react-onclickoutside';
 import Config from './BeaconConfig';
+import TetherComponent from 'react-tether';
 
 import '../assets/style.styl';
 
-const BEACON_HEIGHT = 30;
-const BEACON_WIDTH = 30;
-const TOOLTIP_MARGIN = 10;
 const TOOLTIP_OVERLAY_CLASS = 'tour-overlay';
 const TOOLTIP_FADED_CLASS = 'tour-faded';
 const TARGET_CLONE_ID = 'tour-target-clone';
 
 export class Beacon extends React.Component {
 
   static propTypes = {
-    position: PropTypes.string,
     persistent: PropTypes.bool,
-    children: PropTypes.node
+    tooltipText: PropTypes.string.isRequired,
```

Comment:
The beacon is the parent of the target element now, so we put the text inside an attribute.